XSDElement/src/main/java/com/modelsolv/reprezen/gentemplates/xsdelement/xtend/FeatureHelper.xtend

```diff
@@ -109,71 +102,37 @@ class FeatureHelper {
 		if(feature.maxOccurs != -1) feature.maxOccurs.normalizedMaxOccurs else UNBOUNDED
 	}
 
-	def dispatch Iterable<Feature> getSequenceProperties(Structure dataType) {
-		dataType.ownedFeatures.sequenceProperties.map[it as Feature]
-	}
-
-	def dispatch Iterable<PropertyRealization> getSequenceProperties(ServiceDataResource dataResource) {
-		Iterables.concat(dataResource.type.ownedFeatures.referenceProperties.map[it as PropertyRealization],
-			dataResource.primitiveMultiProperties.map[it as PropertyRealization])
-	}
-
-	def dispatch Iterable<PropertyRealization> getSequenceProperties(TypedMessage message) {
-		Iterables.concat(message.actualType.ownedFeatures.referenceProperties.map[it as PropertyRealization],
-			message.primitiveMultiProperties.map[it as PropertyRealization])
-	}
-
-	def dispatch Iterable<Feature> getSequenceProperties(Iterable<Feature> features) {
-		Iterables.concat(features.referenceProperties.map[it as Feature],
-			features.primitiveMultiProperties.map[it as Feature])
-	}
-
 	def Iterable<PrimitiveProperty> getPrimitiveProperties(Iterable<Feature> features) {
```

Comment:
I like the practice of being explicit about return types for dispatch methods, so that we avoid Xtend's derivation of them with potentially surprising results. I'm not sure the same is helpful for non-dispatch methods, where the actual type is visible by hovering over the method name in eclipse.